packages/cli/src/commands/generate/index.ts

```diff
@@ -15,14 +17,14 @@ export const generateSchema = async ({
     defaultSchema,
     schemas = [],
     tables = [],
+    schemaDefOverrides,
 }: GenerateSchemaOptions) => {
-    const result = await generate({
+    return await generate({
         defaultSchema,
         connectionString,
         includeSchemas: schemas,
         includeTables: tables,
-        outDir: out,
+        schemaDefOverrides: schemaDefOverrides,
```

Comment:
SchemaDefOverrides is a `CLI`-only construct. It shouldn't "leak" into the /introspect package.

We currently only allow synthql.cofig.json, but in the near future we will also allow synthql.config.(js|ts), where we will allow users to maybe functions to override the schemas.

The important bit to take from this is that schemaDefOverrides as a concept is very much dependent on the shape of the synthql.config.json and we shouldn't leak that implementation detail into the /introspect pacakge.

As you also noticed, this also forces us to share a type between these two packages through the /queries package (which has nothing to do with this at all).

So what's the proper fix? Quite simple, actually. See the following pseudocode:

1. Add a property on the generate function that allows for passing an array of table overriders.
```ts
Array<{
    test: (table) => boolean,
    applyOverride: (oldTable) => NewTable
}>
```
These overriders can specify a `test` to determine on which tables they should apply to, and when the test is true, a function that applies the override.

2. Add a function that takes `synthql.config.json` format and creates an array of overriders 
```ts
const overrides = createTableOverrides(jsonConfig)
```